src/index.ts

```diff
@@ -2,14 +2,33 @@ import type { Components } from '@libp2p/interfaces/components'
 import type { StreamMuxerFactory, StreamMuxerInit } from '@libp2p/interfaces/stream-muxer'
 import { MplexStreamMuxer } from './mplex.js'
 
-export interface MplexInit extends StreamMuxerInit {
+export interface MplexInit {
+  /**
+   * The maximum size of message that can be sent in one go in bytes.
+   * Messages larger than this will be split into multiple smaller
+   * messages.
+   */
   maxMsgSize?: number
+
+  /**
+   * The maximum number of multiplexed streams that can be open at any
+   * one time. An attempt to open more than this will throw.
+   */
+  maxStreamsPerConnection?: number
```

Comment:
newbie question here: I assume it's not possible to just put the default here?

Reply:
No, the interface is removed at compile time so it can't have default values.